### ISSUE TITLE
Invalid xml generated by reporter

### ISSUE THREAD

**Issue by Bronek** [catchorg/Catch2#1207]
## Description

The xml reporter is not escaping certain special character in `<Expanded>` section

### Steps to reproduce
Run the following test, using `-r xml` option:
```
#define CATCH_CONFIG_MAIN
#include <catch2/catch.hpp>
#include <cstddef>

using size_type = uint8_t;
constexpr static size_type npos = 255;
size_type foo() { return npos - 1; }

TEST_CASE("Dummy") {
    CHECK(foo() == npos);
}
```

### Extra information

Depending on console/terminal options, the output on console may **not** show special characters. In this is the case, please redirect the output to a file or directly to `less`. For example on my console I can see:
```
$ tests/tests Dummy -r xml
<?xml version="1.0" encoding="UTF-8"?>
<Catch name="tests">
  <Group name="tests">
    <TestCase name="Dummy" filename="/home/LAN/bronek/Sources/market/tests/main.cpp" line="9">
      <Expression success="false" type="CHECK" filename="/home/LAN/bronek/Sources/market/tests/main.cpp" line="10">
        <Original>
          foo() == npos
        </Original>
        <Expanded>
          '' == ''
        </Expanded>
      </Expression>
      <OverallResult success="false"/>
    </TestCase>
    <OverallResults successes="0" failures="1" expectedFailures="0"/>
  </Group>
  <OverallResults successes="0" failures="1" expectedFailures="0"/>
</Catch>
```
The xml **appears** to be well-formed above, but the reason for failing test is shrouded in mystery. The error becomes visible only if I do:

```
$ tests/tests Dummy -r xml | less
```

The error is also visible for succeeding test cases (if `-s` option is used, of course).

* Catch version: **v2.1.2**
* Operating System: **Linux**
* Compiler+version: **GCC 7.3** and **Clang 5.0.1**

**Comment by Bronek** [comment]
I think that my console is simply ignoring an invalid UTF-8 code. In UTF-8 (which is explicitly declared at the start of the document), a codepoint starting with 0xF (i.e. all 4 initial bit set) is expected to be composed of 4 bytes. Here we have single byte starting with 0xF, i.e. invalid UTF-8 character. This character should have been escaped or simply written out as a integral value rather than string.

**Comment by Bronek** [comment]
Fixed with 9171e9b

**Comment by horenmar** [comment]
Not merged to master yet (and I found at least one bug in the process of adding tests).
